packages/gatsby/src/utils/jobs/manager.ts

```diff
@@ -7,62 +7,23 @@ import _ from "lodash"
 import { createContentDigest, slash } from "gatsby-core-utils"
 import reporter from "gatsby-cli/lib/reporter"
 import { IPhantomReporter } from "gatsby-cli"
-
-enum MESSAGE_TYPES {
-  JOB_CREATED = `JOB_CREATED`,
-  JOB_COMPLETED = `JOB_COMPLETED`,
-  JOB_FAILED = `JOB_FAILED`,
-  JOB_NOT_WHITELISTED = `JOB_NOT_WHITELISTED`,
-}
-
-interface IBaseJob {
-  name: string
-  outputDir: string
-  args: Record<string, any>
-}
-
-interface IJobInput {
-  inputPaths: Array<string>
-  plugin: {
-    name: string
-    version: string
-    resolve: string
-  }
-}
-
-interface IInternalJob {
-  id: string
-  contentDigest: string
-  inputPaths: Array<{
-    path: string
-    contentDigest: string
-  }>
-  plugin: {
-    name: string
-    version: string
-    resolve: string
-    isLocal: boolean
-  }
-}
-
+import {
+  JobInput,
+  InternalJob,
+  MESSAGE_TYPES,
+  IJobCreatedMessage,
+  IJobCompletedMessage,
+  IJobFailed,
+  IJobNotWhitelisted,
+  WorkerError,
+} from "./types"
```

Comment:
Shared types moved do `./types.ts`. Also moved `jobs-manager.ts` to `jobs/manager.ts` to accomudate extra `jobs` modules